exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,143 @@
+    } else if (CUMULATIVE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
+    } else {
+      logger.error("Metric type {} not supported", metricType);
+      return null;
+    }
+    if (LONG_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.INT64);
+    } else if (DOUBLE_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+    } else {
+      logger.error("Metric type {} not supported", metricType);
+      return null;
+    }
+    return builder.build();
+  }
+
+  private static String mapMetricType(String instrumentName) {
+    for (String domain : KNOWN_DOMAINS) {
+      if (instrumentName.contains(domain)) {
+        return instrumentName;
+      }
+    }
+    return DESCRIPTOR_TYPE_URL + instrumentName;
+  }
+
+  static LabelDescriptor mapConstantLabel(String key, String value) {
+    LabelDescriptor.Builder builder = LabelDescriptor.newBuilder().setKey(key);
+    if ("true".equalsIgnoreCase(value) || "false".equalsIgnoreCase(value)) {
+      builder.setValueType(LabelDescriptor.ValueType.BOOL);
+    } else if (Longs.tryParse(value) != null) {
+      builder.setValueType(LabelDescriptor.ValueType.INT64);
+    } else {
+      builder.setValueType(LabelDescriptor.ValueType.STRING);
+    }
+    return builder.build();
+  }
+
+  static Point mapPoint(Map<MetricWithLabels, Long> lastUpdatedTime, MetricData metric, MetricWithLabels updateKey,
+      Instant exporterStartTime, long pointCollectionTime) {
+    Builder pointBuilder = Point.newBuilder();
+    Type type = metric.getDescriptor().getType();
+    if (LONG_TYPES.contains(type)) {
+      pointBuilder.setValue(TypedValue.newBuilder().setInt64Value(
+          ((MetricData.LongPoint) metric.getPoints().iterator().next()).getValue()));
+    } else if (DOUBLE_TYPES.contains(type)) {
+      pointBuilder.setValue(TypedValue.newBuilder().setDoubleValue(
+          ((MetricData.DoublePoint) metric.getPoints().iterator().next()).getValue()));
+    } else {
+      logger.error("Type {} not supported", type);
+      return null;
+    }
+    pointBuilder.setInterval(
+        mapInterval(lastUpdatedTime, updateKey, type, exporterStartTime, pointCollectionTime));
+    return pointBuilder.build();
+  }
+
+  static TimeInterval mapInterval(Map<MetricWithLabels, Long> lastUpdatedTime,
+      MetricWithLabels updateKey, Type descriptorType, Instant exporterStartTime, long pointCollectionTime) {
+    long seconds;
+    int nanos;
+    if (CUMULATIVE_TYPES.contains(descriptorType)) {
+      if (!lastUpdatedTime.containsKey(updateKey)) {
+        // The aggregation has not reset since the exporter
+        // has started up, so that is the start time
+        seconds = exporterStartTime.getEpochSecond();
+        nanos = exporterStartTime.getNano();
+      } else {
+        // The aggregation reset the last time it was exported
```

Comment:
I don't think this comment ported well from Python, since Otel Java doesn't have `stateful`. I'm not too familiar with Otel java but my understanding is that for Counter instruments, `export()` receives a Delta for the interval and so the aggregation is "reset" *every* time (it's different in python because of the `stateful` option we discussed). So for Counters , this branch will actually run every export cycle except for the first one when `lastUpdateTime` is missing the key. I talked to @kjordy and this "fake deltas" approach is fine (so long as GCM doesn't allow DELTAs for custom metrics, which would be better).
```suggestion
        // Send "fake deltas" since GCM doesn't support DELTA for custom metrics
```

However, I would imagine that for `SumObserver`, you would still get `MONOTONIC_*` but then the values that `export()` receives would actually already be cumulative. See [here in the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#sumobserver):
> SumObserver is the asynchronous instrument corresponding to Counter, used to capture a monotonic sum with `Observe(sum)`. "Sum" appears in the name to remind users that it is used to capture sums directly.

For SumObservers then, you would actually want to continue using `exporterStartTime` here, IIUC.

Reply:
Thanks for pointing this out! I checked the proto (https://github.com/open-telemetry/opentelemetry-proto/blob/7ae34aaceaeb19ce6e87b2b6ddb542f0104fd72c/opentelemetry/proto/metrics/v1/metrics.proto#L96) and the way https://github.com/open-telemetry/opentelemetry-java/blob/d42d4cd9e2d79c8e5bdc3c43aac5e077f4147794/exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/MetricAdapter.java#L195 is mapping the interval, and it seems like it's safe to use the start time from the Point, which simplifies things by quite a bit. Please let me know if this doesn't look correct

Reply:
I couldn't tell 100% from looking at that code, but that behavior definitely makes sense 👍